cdap-event-writer-spi/src/main/java/io/cdap/cdap/spi/events/PubSubEventReader.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+import java.util.Optional;
+
+/**
+ * {@link PubSubEventReader} implementation for listening for events from Pub/Sub.
+ */
+public interface PubSubEventReader {
+  /**
+   * Method to initialize PubSubEventReader.
+   *
+   * @param context Configuration properties of reader
+   */
+  void initialize(PubSubEventReaderContext context);
+
+  /**
+   * Pull exactly one message from PubSub if available.
+   *
+   * @return Message wrapped in an {@link Optional}
+   */
+  Optional<ReceivedEvent> pull();
+
+  /**
+   * Sends an acknowledgement response to PubSub.
+   *
+   * @param ackId Ack Id of Message to acknowledge
+   * @throws Exception Invalid ACK ID / Expired ACK ID
+   */
+  void ack(String ackId) throws Exception;
```

Comment:
Would it be better to return a boolean representing success or failure of the method?